chains/evm/evmtransaction/evm-tx_test.go

```diff
@@ -0,0 +1,66 @@
+package evmtransaction
+
+import (
+	"math/big"
+	"testing"
+
+	mock_evmgaspricer "github.com/ChainSafe/chainbridge-core/chains/evm/evmgaspricer/mock"
+	"github.com/ChainSafe/chainbridge-core/e2e/evm"
+	"github.com/ethereum/go-ethereum/core/types"
+
+	"github.com/ChainSafe/chainbridge-core/chains/evm/evmgaspricer"
+	"github.com/ethereum/go-ethereum/common"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/suite"
+)
+
+type EVMTxTestSuite struct {
+	suite.Suite
+	client *mock_evmgaspricer.MockLondonGasClient
+}
+
+func TestRunTestSuite(t *testing.T) {
+	suite.Run(t, new(EVMTxTestSuite))
+}
+
+func (s *EVMTxTestSuite) SetupSuite()    {}
+func (s *EVMTxTestSuite) TearDownSuite() {}
+func (s *EVMTxTestSuite) SetupTest() {
+	gomockController := gomock.NewController(s.T())
+	s.client = mock_evmgaspricer.NewMockLondonGasClient(gomockController)
+}
+func (s *EVMTxTestSuite) TearDownTest() {}
+
+func (s *EVMTxTestSuite) TestNewTransactionWithStaticGasPricer() {
+	s.client.EXPECT().SuggestGasPrice(gomock.Any()).Return(big.NewInt(1000), nil)
+	txFabrid := NewTransaction
```

Comment:
```suggestion
	txFabric := NewTransaction
```

Reply:
It is the same in test below that

Reply:
yeap that so funny that I almost died laughing